src/format/multifactor/PasswordAuthenticationFactor.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2024 KeePassXC Team <team@keepassxc.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 or (at your option)
+ * version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PasswordAuthenticationFactor.h"
+#include "keys/PasswordKey.h"
+
+#include <QCryptographicHash>
+#include <QDebug>
+
+PasswordAuthenticationFactor::PasswordAuthenticationFactor(const QSharedPointer<AuthenticationFactor>& factor)
+{
+    m_name = factor->getName();
+    m_keyType = factor->getKeyType();
+    m_keySalt = factor->getKeySalt();
+    m_wrappedKey = factor->getWrappedKey();
```

Comment:
This code is repeated a lot. Why not just store the shared pointer and avoid all the copies?

Reply:
I think each subclass has effectively a copy contructor. In my opinion it wouldn't make sense to store the shared pointer, as that would mean each time you copy the class you would add another proxy to the chain, so calling a method would end up calling a chain of methods all the way up the set of copied classes and copies would use progressively more and more memory.

I don't feel this is too much duplication, personally: each subclass takes the fields it needs from the parent.